src/wifi/model/dmg-sta-wifi-mac.cc

```diff
@@ -3098,8 +3098,10 @@ DmgStaWifiMac::Receive (Ptr<Packet> packet, const WifiMacHeader *hdr)
               if (!beaconInterval.IsDiscoveryMode ())
                 {
                   /* This function is triggered on NanoSeconds basis and Thr duration field is in MicroSeconds basis */
-                  Time dmgBeaconDurationUs = MicroSeconds (ceil (static_cast<double> (m_phy->GetLastRxDuration ().GetNanoSeconds ()) / 1000));
-                  Time startTime = hdr->GetDuration () + (dmgBeaconDurationUs - m_phy->GetLastRxDuration ()) + GetMbifs ();
+                  //Time dmgBeaconDurationUs = MicroSeconds (ceil (static_cast<double> (m_phy->GetLastRxDuration ().GetNanoSeconds ()) / 1000));
+                  //Time startTime = hdr->GetDuration () + (dmgBeaconDurationUs - m_phy->GetLastRxDuration ()) + GetMbifs ();
+                  Time btiDuration = MicroSeconds (operationElement->GetMinBHIDuration ()) - m_abftDuration - m_atiDuration - 2 * GetMbifs ();
+                  Time startTime = btiDuration - (Simulator::Now () - m_biStartTime) + GetMbifs ();
```

Comment:
Using m_biStartTime is a bit like cheating: you are assuming perfect synchronization between the AP and the STA clocks